site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.tsx

```diff
@@ -29,6 +29,13 @@ export const TemplateSettingsPage: FC = () => {
     (data: UpdateTemplateMeta) => updateTemplateMeta(template.id, data),
     {
       onSuccess: async (data) => {
+        // This update has a chance to return a 304 which means nothing was updated.
+        // In this case, the return payload will be empty and we should use the
+        // original template data.
+        if (!data) {
+          data = template;
+        }
```

Comment:
I think this change works for our purposes here. I guess there's just two other things that come to mind:

1. At the end of the day, we do need an `undefined` check, but right now, there's nothing surfacing that the data could be undefined at the type level.

   My worry is that someone will see that we're doing a check for something that appears non-nullable, and not be sure whether the comment still applies. Could you also add an `undefined` union for `updateTemplateMeta`?
   
   I know that we're using this API call in two other files, though, so there might be some edge cases with changing it everywhere. `onSuccess`'s `data` argument could be changed to include `undefined` as a bandaid solution by broadening the type
```ts
export const updateTemplateMeta = async (
  templateId: string,
  data: TypesGen.UpdateTemplateMeta,
): Promise<TypesGen.Template | undefined> => {
  const response = await axios.patch<TypesGen.Template | undefined>(
    `/api/v2/templates/${templateId}`,
    data,
  );
  return response.data;
};
```

2. I haven't dived too deeply into our settings stuff, but my gut feeling is that we don't necessarily need to invalidate the queries and go through a refetch if no data has changed. I wonder if something like this might be good:
```ts
onSuccess: async (data) => {
  if (!data) {
    data = template;
  } else {
    await queryClient.invalidateQueries(
      templateByNameKey(orgId, data.name),
    );
  }
}
```